include/astshim/ChebyMap.h

```diff
@@ -0,0 +1,321 @@
+    minimise the sum of the squared residuals between the sampled values
+    of X and P_i(Y).
+
+    This fitting process is performed repeatedly with increasing
+    polynomial orders (starting with linear) until the target
+    accuracy is achieved, or a specified maximum order is reached. If
+    the target accuracy cannot be achieved even with this maximum-order
+    polynomial, the best fitting maximum-order polynomial is returned so
+    long as its accuracy is better than
+    "maxacc". If it is not, an error is reported.
+
+    @param[in] forward  If true the forward transformation is replaced.
+                    Otherwise the inverse transformation is replaced.
+    @param[in] acc  The target accuracy, expressed as a geodesic distance within
+                    the ChebyMap's input space (if `forward` is false)
+                    or output space (if `forward` is true).
+    @param[in] maxacc  The maximum allowed accuracy for an acceptable polynomial,
+                    expressed as a geodesic distance within the ChebyMap's input space
+                    (if `forward` is false) or output space (if `forward` is true).
+    @param[in] maxorder  The maximum allowed polynomial order. This is one more than the
+                    maximum power of either input axis. So for instance, a value of
+                    3 refers to a quadratic polynomial.
+                    Note, cross terms with total powers greater than or equal to `maxorder`
+                    are not inlcuded in the fit. So the maximum number of terms in
+                    each of the fitted polynomials is `maxorder*(maxorder + 1)/2.`
+    @param[in] lbnd  A vector holding the lower bounds of a rectangular region within
+                    the ChebyMap's input space (if `forward` is false)
+                    or output space (if `forward` is true).
+                    If both lbnd and ubnd are empty (the default) then they will be estimated.
+                    The new polynomial will be evaluated over this rectangle. The length
+                    should equal getNin() or getNout(), depending on `forward`.
+    @param[in] ubnd  A vector holding the upper bounds of a rectangular region within
+                    the ChebyMap's input space (if `forward` is false)
+                    or output space (if `forward` is true).
+                    If both lbnd and ubnd are empty (the default) then they will be estimated.
+                    The new polynomial will be evaluated over this rectangle. The length
+                    should equal getNin() or getNout(), depending on `forward`.
+
+    @throw std::invalid_argument if the size of `lbnd` or `ubnd` does not match getNin() (if `forward` false)
+                    or getNout() (if `forward` true).    
+
+    The variant that takes omits the `lbnd` and `ubnd` arguments
+    uses the full domain of the polynomial whose inverse is being fit.
+    @{
+    */
```

Comment:
This will cause two methods to be independently documented with the text above (including the confusing "the variant" string that doesn't point to anything). You'll also get a doxygen warning for the nonexistent `lbnd` and `ubnd` parameters on the second method, and a confusing reference to a nonexistent default value for the first one.

I'd recommend an explicit cross-reference like you had me do for `SpherePoint`, e.g., "This method is the same as [link], except that the lower and upper bounds will be estimated."